bindings/python/bergamot.cpp

```diff
@@ -0,0 +1,227 @@
+             return std::string(view.data(), view.size());
+           })
+      .def("sentence",
+           [](const AnnotatedText &annotatedText,
+              size_t sentenceIdx) -> std::string {
+             auto view = annotatedText.sentence(sentenceIdx);
+             return std::string(view.data(), view.size());
+           })
+      .def("wordAsByteRange", &AnnotatedText::wordAsByteRange)
+      .def("sentenceAsByteRange", &AnnotatedText::sentenceAsByteRange)
+      .def_readonly("text", &AnnotatedText::text);
+
+  py::class_<Response>(m, "Response")
+      .def(py::init<>())
+      .def_readonly("source", &Response::source)
+      .def_readonly("target", &Response::target)
+      .def_readonly("alignments", &Response::alignments);
+
+  py::bind_vector<std::vector<std::string>>(m, "VectorString");
+  py::bind_vector<std::vector<Response>>(m, "VectorResponse");
+
+  py::enum_<ConcatStrategy>(m, "ConcatStrategy")
+      .value("FAITHFUL", ConcatStrategy::FAITHFUL)
+      .value("SPACE", ConcatStrategy::SPACE)
+      .export_values();
+
+  py::class_<ResponseOptions>(m, "ResponseOptions")
+      .def(py::init<>([](bool qualityScores, bool alignment, bool HTML,
+                         bool sentenceMappings, ConcatStrategy strategy) {
+             return ResponseOptions{qualityScores, alignment, HTML,
+                                    sentenceMappings, strategy};
+           }),
+           py::arg("qualityScores") = true, py::arg("alignment") = false,
+           py::arg("HTML") = false, py::arg("sentenceMappings") = true,
+           py::arg("concatStrategy") = ConcatStrategy::FAITHFUL)
+      .def_readwrite("qualityScores", &ResponseOptions::qualityScores)
+      .def_readwrite("HTML", &ResponseOptions::HTML)
+      .def_readwrite("alignment", &ResponseOptions::alignment)
+      .def_readwrite("concatStrategy", &ResponseOptions::concatStrategy)
+      .def_readwrite("sentenceMappings", &ResponseOptions::sentenceMappings);
+
+  py::class_<ServicePyAdapter>(m, "Service")
+      .def(py::init<const Service::Config &>())
+      .def("modelFromConfig", &ServicePyAdapter::modelFromConfig)
+      .def("modelFromConfigPath", &ServicePyAdapter::modelFromConfigPath)
+      .def("translate", &ServicePyAdapter::translate)
+      .def("pivot", &ServicePyAdapter::pivot);
+
+  py::class_<Service::Config>(m, "ServiceConfig")
+      .def(py::init<>([](size_t numWorkers, bool cacheEnabled, size_t cacheSize,
+                         size_t cacheMutexBuckets, std::string logging) {
+             Service::Config config;
+             config.numWorkers = numWorkers;
+             config.cacheEnabled = cacheEnabled;
+             config.cacheSize = cacheSize;
+             config.cacheMutexBuckets = cacheMutexBuckets;
+             config.logger.level = logging;
+             return config;
+           }),
+           py::arg("numWorkers") = 1, py::arg("cacheEnabled") = false,
+           py::arg("cacheSize") = 20000, py::arg("cacheMutexBuckets") = 1,
```

Comment:
Does `cacheSize = 20000` but `cacheEnabled = false` entail that it does allocate space for 20k entries, but never use it by default? Is `cacheMutexBuckets=1` a good default if someone were to change the `numWorkers`?

Reply:
So, w.r.t `cacheMutexBuckets`, I want to stop exposing them to the user, same as what translateLocally is doing. As for the `cacheEnabled` what do you think of wrapping cache in `std::optional`? It is also possible to set a per request enabled/disabled as well even if we've created the cache-store. 

In any case, choosing to defer this for later.

Reply:
> So, w.r.t cacheMutexBuckets, I want to stop exposing them to the user, same as what translateLocally is doing. 

I think this is a good idea. Like, there has to be a theoretical optimum value with respect to the number of threads and the cacheSize. bergamot-translator can figure out that relation. That shouldn't be up to the user.

> As for the cacheEnabled what do you think of wrapping cache in std::optional? It is also possible to set a per request enabled/disabled as well even if we've created the cache-store.

My problem is that you allocate a 20.000 ptr vector by default, even when cacheEnabled is false. Easiest would be to just add a ternary operator [here](https://github.com/browsermt/bergamot-translator/blob/6445a45b2538adc2fa6e40d77d29498c7a962085/src/translator/service.cpp#L125) to set cacheSize to 0 and save a dynamic allocation for it when cacheEnabled is off.

Other options would be making cache a Ptr (so you can manage it on your own, like clearing it when switching models in translateLocally since you know that old entries wont be very relevant at that point, or keep it between AsyncService changes when changing number of workers or something) or keep it like it is now, but having cacheSize=0 shortcut all calls into it or something. Or indeed wrap it in a std::optional to save it from being initialised when cacheEnabled=0.